ambry-store/src/main/java/com.github.ambry.store/IndexSegment.java

```diff
@@ -254,6 +265,23 @@ long getLastModifiedTime() {
   }
 
   /**
+   * The version of the {@link PersistentIndex} that this {@link IndexSegment} is based on
+   * @return the version of the {@link PersistentIndex} that this {@link IndexSegment} is based on
+   */
+  short getVersion() {
+    return this.version;
```

Comment:
MockIndexSegmentV0 overrides this method. If not, we have an option to make this variable protected and override in MockIndexSegmentV0.